res/controllers/Denon-MC7000-scripts.js

```diff
@@ -0,0 +1,859 @@
+};
+
+// Sensitivity of the jog wheel (also depends on audio latency)
+MC7000.jogParams = {
+    // Lower values for less, higher values for more sensitive
+    jogSensitivity: 30, // default: 30
+    // this will limit the parameter of "jog" (keep between 0.5 and 3)
+    maxJogValue: 3      // default: 3
+};
+
+/*/////////////////////////////////
+//      USER VARIABLES END       //
+/////////////////////////////////*/
+
+
+/* OTHER VARIABLES - DONT'T TOUCH EXCEPT YOU KNOW WHAT YOU DO */
+
+// Resolution of the jog wheel, set so the spinny
+// Jog LED to match exactly the movement of the Jog Wheel
+// The physical resolution seems to be around 1100
+MC7000.jogWheelTicksPerRevolution = 894;
+
+// must be "true" for Needle Search to be active
+MC7000.needleSearchTouched = [true, true, true, true];
+
+// initial value for VINYL mode per Deck (see above for user input)
+MC7000.isVinylMode = [MC7000.VinylModeOn, MC7000.VinylModeOn, MC7000.VinylModeOn, MC7000.VinylModeOn];
+
+// used to keep track of which the rateRange of each slider.
+// value used as an index to MC7000.rateRanges
+MC7000.currentRateRangeIndex = [0, 0, 0, 0];
+
+// initialize the "factor" function for Spinback
+MC7000.factor = [];
+
+// initialize the PAD Mode to Hot Cue and all others off when starting
+MC7000.PADModeCue = [true, true, true, true];
+MC7000.PADModeCueLoop = [false, false, false, false];
+MC7000.PADModeFlip = [false, false, false, false];
+MC7000.PADModeRoll = [false, false, false, false];
+MC7000.PADModeSavedLoop = [false, false, false, false];
+MC7000.PADModeSlicer = [false, false, false, false];
+MC7000.PADModeSlicerLoop = [false, false, false, false];
+MC7000.PADModeSampler = [false, false, false, false];
+MC7000.PADModeVelSamp = [false, false, false, false];
+MC7000.PADModePitch = [false, false, false, false];
+
+// Define the MIDI signal for red LED at VU Meters
+MC7000.VuMeterLEDPeakValue = 0x76;
+
+// PAD Mode Colors
+MC7000.padColor = {
+    "alloff": 0x01,         // switch off completely
+    // Hot Cue
+    "hotcueon": 0x04,       // darkblue Hot Cue active
+    "hotcueoff": 0x02,      // lightblue Hot Cue inactive
+    // Cue Loop
+    "cueloopon": 0x0D,      // Cueloop colour for activated cue point
+    "cueloopoff": 0x1A,     // Cueloop colour inactive
+    // Roll
+    "rollon": 0x20,         // BeatloopRoll active colour
+    "rolloff": 0x06,        // BeatloopRoll off colour
+    // Slicer
+    "sliceron": 0x11,       // activated Slicer
+    "slicerJumpFwd": 0x31,  // Sliver forward jump
+    "slicerJumpBack": 0x31, // Sliver backward jump
+    // Sampler
+    "samplerloaded": 0x38,  // dark pink Sampler loaded colour
+    "samplerplay": 0x09,    // green Sampler playing
+    "sampleroff": 0x12     // light pink Sampler standard colour
+};
+
+/* DECK INITIALIZATION */
+MC7000.init = function() {
+
+    // Decks
+    MC7000.leftDeck = new MC7000.Deck(1, 3);
+    MC7000.rightDeck = new MC7000.Deck(2, 4);
```

Comment:
This is unnecessary. `MC7000.Deck` does no really create a meaningful object, and you never access it. The only reason why this is currently needed is because you *assign global function from inside the constructor*.
By calling the constructor twice, you even assign these functions twice (the second call overrides the functions assigned by the first).

Reply:
no idea - a sample code was like this in the wiki and also in the Denon MC4000 script so I thought it is needed.